mplc/scenario.py

```diff
@@ -174,6 +174,11 @@ def __init__(
         # (% of samples of the dataset for each partner, ...
         # ... has to sum to 1, and number of items has to equal partners_count)
         self.amounts_per_partner = amounts_per_partner
+        if np.sum(self.amounts_per_partner) != 1:
+            raise ValueError("The sum of the amount per partners you provided isn't equal to 1")
+        if len(self.amounts_per_partner) != self.partners_count:
+            raise AttributeError(f"The amounts_per_partner list should have a size ({len(self.amounts_per_partner)}) "
+                                 f"equals to partners_count ({self.partners_count})")
```

Comment:
I don't see why you want to move those from the splitter to the scenario ?
I think it might create issues if someone is using a splitter outside a scenario

Reply:
That is true, it could. However, this shouldn't happen too often that someone wants to do that.

The reason I did that is that some splitters might not require those assertions, and might even be incompatible with them. That is the case of the `FlexibleSplitter` which works without any `amounts_per_partner`.

Reply:
If you see a more direct and pythonic way for that, don't hesitate that would be good!

Reply:
Now thinking about it, I think those asserts (or similar ones) should be **both** in `Scenario` and in `Splitter`:

- at scenario level, for a `Scenario` to be coherent, those tests need to pass otherwise the user config doesn't add up
- at splitter level, some splitters need to enforce that too, and, as you say @arthurPignet , in some cases one might want to use a splitter outside a scenario

The 2 perspectives complement each other, but both need to live independently too.

Reply:
So I added an abstract method for coherence test at the beginning of the `split()` function. Each Splitter has to implement its own coherence tests within it (although 2 of them are generic enough to be implemented within the abstract class)